plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1504,6 +1507,43 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         metastore.dropTable(session, handle.getSchemaName(), handle.getTableName());
     }
 
+    @Override
+    public void renameTable(ConnectorSession session, ConnectorTableHandle tableHandle, SchemaTableName newTable)
+    {
+        DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
+        Table table = metastore.getTable(handle.getSchemaName(), handle.getTableName()).orElseThrow(() -> new TableNotFoundException(handle.getSchemaTableName()));
+
+        if (table.getTableType().equals(MANAGED_TABLE.name())) {
+            if (!renameManagedTablesEnabled) {
+                throw new TrinoException(NOT_SUPPORTED, "Renaming managed tables will move the data on the underlying storage system. Set 'delta.rename-managed-tables.enabled' to enable this feature.");
+            }
+
+            Database schema = metastore.getDatabase(newTable.getSchemaName()).orElseThrow(() -> new SchemaNotFoundException(newTable.getSchemaName()));
+            Optional<String> schemaLocation = getSchemaLocation(schema);
+            if (schemaLocation.isEmpty()) {
+                throw new TrinoException(NOT_SUPPORTED, "The 'location' property must be specified either for the table or the schema");
+            }
+            String location = new Path(schemaLocation.get(), newTable.getTableName()).toString();
+            Path targetPath = new Path(location);
```

Comment:
I'd flip these to avoid the double Path construction, alternatively inline the `toString` calls
```suggestion
            Path targetPath = new Path(schemaLocation.get(), newTable.getTableName());
            String location = targetPath.toString();
```